classes/profile.php

```diff
@@ -268,17 +268,21 @@ protected function check_update_userid(int $currentid) : bool {
             return true;
         }
 
-        // If current userid isn't guest, but stored user is guest, update it.
+        if ($currentid === $stored) {
+            return false;
+        }
+
+        // If stored user is guest, update it.
         $guestid = (int) $CFG->siteguest;
-        if ($guestid && $currentid !== $guestid && $stored === $guestid) {
+        if ($guestid && $stored === $guestid) {
             return true;
         }
 
-        // If current user isn't cli admin, but stored user is cli admin, update it.
-        $adminid = get_admin()->id;
-        if ($adminid && $currentid !== $adminid && $stored === $adminid) {
+        // If stored user is cli admin, update it.
+        if (!empty($CFG->siteadmins) && in_array($stored, explode(',', $CFG->siteadmins))) {
```

Comment:
Since the stored value will always (by this stage) be an _int_. Could we simplify this further and based on current experience, assume there is always going to be a siteadmin value of some sort set (as it would be most of the time), and go straight for the explode?

For example:
```suggestion
        if (in_array($stored, explode(',', $CFG->siteadmins))) {
```

If siteadmins could be ever be null, then perhaps it can stay in its current form.

Reply:
According to the comments inside `get_admin()`, siteadmins can be null during unit tests.